### PR TITLE
[RM] Fix windows installer for 8.13

### PR DESCRIPTION
**Pull request by gares**
- [ ] port to master once finished

### CODE REVIEW

**Review by Zimmi48 — APPROVED** [review]
:sparkles:

**Review by MSoegtropIMC — APPROVED** [review]
Yes, the name of the Coq project file changed from UniCoq 1.3 to 1.5.

**Comment by MSoegtropIMC** [comment]
P.S.: UniCoq passes in CI which can be seen from the fact that there is a .finished file:

```
C:\ci\cygwin64_17057_22701\build\flagfiles\unicoq-3dd7abd589eadb09d8eed8574ca3ec981ee7fb1f.finished
C:\ci\cygwin64_17057_22701\build\flagfiles\unicoq-3dd7abd589eadb09d8eed8574ca3ec981ee7fb1f.started
C:\ci\cygwin64_17057_22701\build\flagfiles\vst-ebcd978765a8211d545c8efd2114dec75a991720.started
```

But VST fails (there is a .started but no .finished file).

**Comment by gares** [comment]
Hum, I need help for this one:

> Makefile:125: *** COMPCERT VERSION MISMATCH: COMPCERT_VERSION=version=3.7 but C:/ci/coq64_17447_7515/lib/coq//user-contrib/compcert/VERSION=version=3.8.  Stop.

**Comment by MSoegtropIMC** [comment]
@gares: Indeed afaik CompCert 3.8 is quite new and there is no VST for it as yet. One could try patching the version check and see if it works.

If you have a few days I can look into this and create a new opam package.

**Comment by MSoegtropIMC** [comment]
There is no opam package for compcert 3.8 either, btw.

**Comment by gares** [comment]
Here I'm building with the hashes we use for CI. So something must be wrong/off here (AFAICT CI worked on linux)

**Comment by gares** [comment]
Ok, this is the thing: CI builds with `export COMPCERT=bundled` but I think we don't want this for the installer nor the platform.

**Comment by MSoegtropIMC** [comment]
> Ok, this is the thing: CI builds with export COMPCERT=bundled but I think we don't want this for the installer nor the platform.

No definitely not. Not sure what the build procedure on master is - possibly it disables a lot of version dependency checks, which is appropriate on master but not for a release build. A Coq release should use either CompCert 3.7 and VST 2.6 or in case we can make it in time CompCert 3.8 and VST 2.7 (the latter does not exist as yet). It is more important to use a proper release than to use something which is latest. Usually Andrew is quick with this, but it will take a few days. I will discuss it with him.

**Comment by MSoegtropIMC** [comment]
For the time being I would use CompCert 3.7 and VST 2.6, unless these don't build with 8.13. I will work with Andrew on getting 3.8 and VST 2.7 soon, so that this can be changed before the release.

**Comment by MSoegtropIMC** [comment]
@gares: it is likely that there will be a new VST release until the scheduled 8.13 beta release date, but likely not this week. I could make a VST 2.7 beta release compatible with CompCert 3.8, though, in case this helps.

**Comment by gares** [comment]
Great, I will open an issue tomorrow as per the script I posted today.

**Comment by gares** [comment]
FTR: this script: https://github.com/coq/coq/pull/13449

**Comment by gares** [comment]
The good news is that compcert compiled fine, VST failed as expected. So I'm removing it temporarily to be able to play with the artifact.

**Comment by gares** [comment]
Now it's interval which is broken, and we don't test it on the regular CI. I'll add it.

**Comment by gares** [comment]
For reference: https://github.com/coq/coq/pull/13467

**Comment by gares** [comment]
now elpi fails, I guess it is about the sha1&co... I'll deal with it tomorrow

**Comment by gares** [comment]
@MSoegtropIMC we are getting there, but what I just noticed is that
https://coq.gitlab.io/-/coq/-/jobs/876988114/artifacts/artifacts/buildlogs/coq-local-configure_log.txt
does not build CoqIDE and the script does not fail!
(well, maybe it fails at NSIS time...)

Anyway @coq/coqide-maintainers which is the minimal GTK/lablgtk version we need?

**Comment by gares** [comment]
Sorry, it's written in the config message... it's almost the weekend ;-)

File: dev/build/windows/makecoq_mingw.sh [hunk 1]
```diff
@@ -1955,7 +1955,13 @@ function make_addon_gappa {
   if build_prep_overlay gappa_plugin ; then
     installer_addon_section gappa "Gappa plugin" "Coq plugin for the Gappa tool" ""
     log1 autoreconf
+
+    # coqc - config prints broken paths
+    OLD_OCAMLFIND="$OCAMLFIND"
+    export OCAMLFIND=ocamlfind
     logn configure ./configure
+    export OCAMLFIND="$OLD_OCAMLFIND"
```
Comment (silene) [inline dev/build/windows/makecoq_mingw.sh:1963]: If you do not want to play this dance with environment variables, just pass it on the command line:

    logn configure ./configure OCAMLFIND=ocamlfind

Comment (gares) [reply dev/build/windows/makecoq_mingw.sh:1963]: thanks, I'll do. Unfortunately I did not manage to test your new commit yet, since CoqIDE is broken.

**Comment by gares** [comment]
I close this one. The windows script part of CI has to be removed and replaced by a script using the platform.

**Pull request closed without merging**